main.tf

```diff
@@ -61,6 +79,29 @@ resource "azurerm_storage_account" "example" {
   location                 = azurerm_resource_group.example.location
   account_tier             = "Standard"
   account_replication_type = "GRS"
+
+  queue_properties {
+    hour_metrics {
+      version               = "1.0"
+      include_apis          = true
+      retention_policy_days = 365
+      enabled               = true
+    }
+    minute_metrics {
+      enabled               = true
+      version               = "1.0"
+      include_apis          = true
+      retention_policy_days = 365
+    }
+    logging {
+      delete                = true
```

Comment:
**Tenable.cs remediation**
 
**[MEDIUM]: Ensure that 'Data encryption' is set to 'On' on a SQL Database**
 
**Vulnerability**
Description:

Enable Transparent Data Encryption on every SQL server.

Rationale:

Azure SQL Database transparent data encryption helps protect against the threat of malicious activity by performing real-time encryption and decryption of the database, associated backups, and transaction log files at rest without requiring changes to the application.


 
**Remediation**
**From Azure Portal**

1. Go to 'SQL databases'
2. For each DB instance
3. Click on 'Transparent data encryption'
4. Set 'Data encryption' to 'On'

**From Azure CLI**

Use the below command to enable 'Transparent data encryption' for SQL DB instance.


az sql db tde set --resource-group <resourceGroup> --server <dbServerName> --database <dbName> --status Enabled


**Note:**

- TDE cannot be used to encrypt the logical master database in SQL Database. The master database contains objects that are needed to perform the TDE operations on the user databases.

- Azure Portal does not show master databases per SQL server. However, CLI/API responses will show master databases.

source: tenablecs-